railties/test/application/rake_test.rb

```diff
@@ -40,7 +40,7 @@ def test_not_protected_when_previous_migration_was_not_production
       with_rails_env "test" do
         rails "generate", "model", "product", "name:string"
         rails "db:create", "db:migrate"
-        output = Dir.chdir(app_path) { rails("db:test:prepare", "test") }
+        output = rails("db:test:prepare")
```

Comment:
Again, I think this is important to ensure `test` actually works, because our only assertion is a loose non-match.

It's really not good to make changes to tests just because you don't think something needs to be there.

Reply:
@matthewd do we have to return back execution of `bin/rails test` that was removed by https://github.com/rails/rails/pull/30520/files#diff-e1c35224a4df7f73d90a9a90e45aab2bL32?

Reply:
No. That one is never reached, because the prepare fails. And the following assertion is positive ("this must be there"), not negative ("this specific string should not be").

Reply:
I deleted the changes